src/CQRS/LeanCode.CQRS.AspNetCore/Registration/CQRSObjectsRegistrationSource.cs

```diff
@@ -10,17 +10,32 @@ internal class CQRSObjectsRegistrationSource
     private readonly TypesCatalog contractsCatalog;
     private readonly TypesCatalog handlersCatalog;
 
-    public IEnumerable<CQRSObjectMetadata> Objects { get; }
+    public IEnumerable<CQRSObjectMetadata> Objects { get; private set; }
 
     public CQRSObjectsRegistrationSource(TypesCatalog contractsCatalog, TypesCatalog handlersCatalog)
     {
         this.contractsCatalog = contractsCatalog;
         this.handlersCatalog = handlersCatalog;
 
-        Objects = FindCQRSObjects();
+        Objects = FindCQRSObjects(contractsCatalog, handlersCatalog);
     }
 
-    private IEnumerable<CQRSObjectMetadata> FindCQRSObjects()
+    public IEnumerable<CQRSObjectMetadata> AddAdditionalCQRSObjects(
```

Comment:
I'm not sure about that, IMO we should not allow to modify the registration source. @marcin-dardzinski , what is your opinion on that?

Reply:
It's not an intrinsic problem - as long as we do it before we build routes. We just need to make sure no object is registered twice. 

I think this would be easier to achieve then to have injection points for `TypesCatalog`.

Reply:
We decided to make registration source modifiable and expose it in `CQRSServicesBuilder` so that clients could extend it easily.

Client setup will look like this
```
services.AddCQRS(...).AddForceUpdate();
```